### DISCUSSION TITLE
How to upload invocies for shipments

### DISCUSSION THREAD

**Discussion by iAccidentallyGitHub** [abuzuhri/Amazon-SP-API-CSharp#561] [General]
Hello,

I am trying to use your NuGet package to upload invoices for orders we've shipped. 
There are two things I don't understand.
1. The request only has 3 fields:
 - MarketplaceId - (in my case the ID of Germany)
 - ContentMD5Value - MD5 checksum of the invoice content
 - InvoiceContent - invoice file as a byte array

What I don't understand is, how will Amazon know which order/shipment this invoice belongs to? There is no order id to be filled. Presumably the authentication information will tell amazon which seller I am, but there is no information on even which of our customers this pertains to, let alone which one of their orders.

2. The documentation for the invoice upload endpoint:
https://developer-docs.amazon.com/sp-api/docs/shipment-invoicing-api-v0-reference
says the following: 
Retrieve shipment invoice information in the Brazil marketplace for a selling partner’s Fulfillment by Amazon (FBA) orders.
Does this mean this functionality is only supported in Brazil? Or did they first tested this for Brazil, then released it everywhere and forgot to update the docs?

**Comment by zinsmeik** [comment]
check https://github.com/abuzuhri/Amazon-SP-API-CSharp/issues/546#issuecomment-1618288567
this should work
